tests/unit/test-util/test.adb.js

```diff
@@ -0,0 +1,896 @@
+/* @flow */
+
+import EventEmitter from 'events';
+
+import chai from 'chai';
+import chaiAsPromised from 'chai-as-promised';
+import {describe, it} from 'mocha';
+import sinon from 'sinon';
+
+import {
+  UsageError,
+  WebExtError,
+} from '../../../src/errors';
+import ADBUtils from '../../../src/util/adb';
+
+const fakeADBPackageList = `
+package:org.mozilla.fennec
+package:org.mozilla.firefox
+package:org.some.other.software
+`;
+
+// NOTE: fake /proc/net/unix output format based on the output collected from
+// an android system.
+const fakeSocketFilePrefix = (
+  '00000000: 00000003 00000000 00000000 0001 03  1857'
+);
+
+const fakeRDPUnixSocketFile = (
+  '/data/data/org.mozilla.firefox/firefox-debugger-socket'
+);
+
+const fakeUnixSocketFiles = (`
+${fakeSocketFilePrefix} /dev/socket/mdns
+${fakeSocketFilePrefix}  ${fakeRDPUnixSocketFile}
+`);
+
+// NOTE: fake 'pm dump <APK>' output related to the granted permissions for an
+// android application.
+const fakeAndroidGrantedPermissions = `
+android.permission.READ_EXTERNAL_STORAGE: granted=true
+android.permission.WRITE_EXTERNAL_STORAGE: granted=true
+`;
+
+// Enable chai-as-promised plugin.
+chai.use(chaiAsPromised);
```

Comment:
Is this making a global change to the test suite? If so, I think it would be better in [`tests/unit/runner.js`](https://github.com/mozilla/web-ext/blob/master/tests/unit/runner.js) -- would that work?